src/sentry_scope.h

```diff
@@ -55,9 +55,10 @@ void sentry__scope_cleanup(void);
 
 /**
  * This will notify any backend of scope changes, and persist session
- * information to disk.
+ * information to disk. This function must be called while holding the scope
+ * lock, and it will be unlocked internally.
```

Comment:
that seems a bit easy to get wrong, but ok.  i guess it'll just crash and not even pass the test suite if you get it wrong?

Reply:
I don’t really know what happens when you try to unlock an unlocked lock :-D The only use of that function is from within the scope macro.